status/views.py

```diff
@@ -0,0 +1,37 @@
+from datetime import datetime
+
+from django.db.models import Q
+from django.http import JsonResponse
+from rest_framework import status
+from rest_framework.generics import ListAPIView, get_object_or_404
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from status.models import UserStatus
+from status.serializers import UserStatusSerializer
+from workspace.models import Workspace
+
+
+class UserStatusView(ListAPIView):
+    queryset = UserStatus.objects.prefetch_related('workspace')
+    serializer_class = UserStatusSerializer
+
+    def get_queryset(self, f=None):
+        return self.queryset.filter(f)
+
+    def get(self, request: Request, *args, **kwargs):
+        workspace_hashed_value = kwargs.get('workspace_hashed_value', None)
+        if workspace_hashed_value is None:
+            return JsonResponse(data={'msg': 'no workspace_hashed_value'}, status=status.HTTP_400_BAD_REQUEST)
+        else:
+            # Check `BaseWorkspaceSerializer` is valid.
+            get_object_or_404(Workspace, hashed_value=workspace_hashed_value)
```

Comment:
If we use a shortcut, get_object_or_404, then we don't have to check if workspace_hased_value is "None" or not?
or do we have to double-check that?

Reply:
To check about `workspace_hased_value` literally check that *string value is included in `workspace_hased_value` and not `None`*.

Checking out the object about `Workspace` is getting data from DB.

So that are totally different.

Reply:
I got it, so the first checking is about the kwargs does contain value, => from args
and second check is there is in db => from db data
Thanks you for teaching